nova_plugin/server.py

```diff
@@ -387,6 +389,7 @@ def _set_network_and_ip_runtime_properties(server):
     ctx.instance.runtime_properties[IP_PROPERTY] = manager_network_ip
 
 
+@retry(stop_max_attempt_number=10)
```

Comment:
Don't use `retry`
instead, withing the operation, instead of raising `NonRecoverableError` do 
``` python
return ctx.operation.retry(message='...')
```